cluster-autoscaler/cloudprovider/aws/aws_cloud_provider.go

```diff
@@ -139,6 +139,13 @@ func (aws *awsCloudProvider) HasInstance(node *apiv1.Node) (bool, error) {
 		return true, cloudprovider.ErrNotImplemented
 	}
 
+	// nodes that belong to an asg that is not autoscaled will not be found in the asgCache below
+	// do not spam warnings about them being unable to be found
+	// users that see this error can add an annotation to their nodes to avoid it
+	if node.Annotations != nil && node.Annotations["k8s.io/cluster-autoscaler/enabled"] == "false" {
```

Comment:
Can we add a test confirming this works correctly?

Reply:
yes, I can add tests if anyone with merge permission can confirm this is a good approach

Reply:
This seems fine to me, yes.

Rather: imo the _ideal_ approach would be to just have that error be logged at a lower verbosity, but I think that's going to be non-trivial to accomplish given how the code is currently organized, so this seems fine as an alternate solution.